app/controllers/blocks_controller.rb

```diff
@@ -0,0 +1,93 @@
+class BlocksController < ApplicationController
+  before_action :login_required
+  before_action :find_block, only: [:edit, :update, :destroy]
+  before_action :require_permission, only: [:edit, :update, :destroy]
+  before_action :editor_setup, only: [:new, :edit]
+
+  def index
+    @page_title = "Blocked Users"
+    @blocks = Block.where(blocking_user: current_user)
+  end
+
+  def new
+    @page_title = "Block User"
+    @block = Block.new
+    @users = []
+    @block.blocking_user = current_user
+  end
+
+  def create
+    @block = Block.new(permitted_params)
+    @block.blocking_user = current_user
+    @block.blocked_user_id = params.fetch(:block, {})[:blocked_user_id]
+
+    if @block.save
+      flash[:success] = "User blocked!"
+      redirect_to blocks_path
+    else
+      flash.now[:error] = {}
+      flash.now[:error][:message] = "User could not be blocked."
+      flash.now[:error][:array] = @block.errors.full_messages
+      @users = [User.find_by(id: params.fetch(:block, {})[:blocked_user_id])]
+      @page_title = 'Block User'
+      render :action => :new
+    end
+  end
+
+  def edit
+    @page_title = 'Edit Block: ' + @block.blocked_user.username
+  end
+
+  def update
+    if @block.update(permitted_params)
+      flash[:success] = "Block updated!"
+      redirect_to blocks_path
+    else
+      flash.now[:error] = {}
+      flash.now[:error][:message] = "Block could not be saved."
+      flash.now[:error][:array] = @block.errors.full_messages
+      @page_title = 'Edit Block: ' + @block.blocked_user.username
+      render :action => :edit
+    end
+  end
+
+  def destroy
+    if @block.destroy
+      flash[:success] = "User unblocked."
+    else
+      flash[:error] = {}
+      flash[:error][:message] = "User could not be unblocked."
+      flash[:error][:array] = @block.errors.full_messages
+    end
+    redirect_to blocks_path
+  end
+
+  private
+
+  def permitted_params
+    params.fetch(:block, {}).permit(:block_interactions, :hide_me, :hide_them)
```

Comment:
Any reason not to include blocking_user_id here? Just because of edit or?

Reply:
because we don't want to allow that on edit, yeah